### DISCUSSION TITLE
Multi-Attach error for volume "pvc-80c52f0e-8ebc-4f29-b93a-12bd0978be8b" Volume is already exclusively attached to one node and can't be attached to another

### DISCUSSION THREAD

**Discussion by tomsunyu** [longhorn/longhorn#5369] [Q&A]
My statefulset mount volumes by longhorn, but my pod of statefulset is always containercreating when I delete the pod. I view the pod logs. The following error as this:

```
Multi-Attach error for volume "pvc-80c52f0e-8ebc-4f29-b93a-12bd0978be8b" Volume is already exclusively attached to one node and can't be attached to another.
```
Why ?

**Comment by laemm-line** [comment]
I got the same error today, it seems like it was caused by a stale VolumeAttachment (`k get volumeattachment`).
Managed to fix it after manually deleting the old VolumeAttachment.

**Comment by weizhe0422** [comment]
`VolumeAttachment` objects are created as CSI-attachers to capture the intent of volume attach/detach to target nodes.

After removing a POD, the AD controller will wait 6 minutes (i.e. maxWaitForUnmountDuration) to force the attached volume to be detached from the node. The `VolumeAttachment` object is then recreated on the next target node.

Therefore, I think you can manually delete the `VolumeAttachment` object to improve recovery time from multiple attach errors.

**Reply by weizhe0422** [reply]
And could you also provide the StatefulSet yaml and Kubernetes version that you used?

I think there is efficiency improvement with informer instead of periodically querying each `VolumeAttachment` object after v1.18.2. 
ref. https://github.com/kubernetes/kubernetes/issues/86281
